cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/anomaly/MultivariateAnomalyDetection.scala

```diff
@@ -635,3 +647,112 @@ class SimpleDetectMultivariateAnomaly(override val uid: String) extends Model[Si
+
+class DetectLastMultivariateAnomaly(override val uid: String) extends CognitiveServicesBase(uid)
+  with HasInternalJsonOutputParser with TimeConverter with HasTimestampCol
+  with HasSetLocation with HasCognitiveServiceInput with HasBatchSize
+  with ComplexParamsWritable with Wrappable
+  with HasErrorCol with SynapseMLLogging with DetectMAParams {
+  logClass()
+
+  def this() = this(Identifiable.randomUID("DetectLastMultivariateAnomaly"))
+
+  def urlPath: String = "anomalydetector/v1.1/multivariate/models/"
+
+  val inputVariablesCols = new StringArrayParam(this, "inputVariablesCols",
+    "The names of the input variables columns")
+
+  def setInputVariablesCols(value: Array[String]): this.type = set(inputVariablesCols, value)
+
+  def getInputVariablesCols: Array[String] = $(inputVariablesCols)
+
+  override def setBatchSize(value: Int): this.type = {
+    logWarning("batchSize should be equal to 1 sliding window.")
+    set(batchSize, value)
+  }
+
+  setDefault(batchSize -> 300)
+
+  override protected def prepareUrl: Row => String = {
+    row: Row => getUrl + s"$getModelId:detect-last"
+  }
+
+  protected def prepareEntity: Row => Option[AbstractHttpEntity] = { row =>
+    val timestamps = row.getAs[Seq[String]](s"${getTimestampCol}_list")
+    val variables = getInputVariablesCols.map(
+      variable => Variable(timestamps, row.getAs[Seq[Double]](s"${variable}_list"), variable))
+    Some(new StringEntity(
+      DLMARequest(variables, getTopContributorCount).toJson.compactPrint
+    ))
+  }
+
+  // scalastyle:off null
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    logTransform[DataFrame]({
+      // check model status first
+      MADUtils.checkModelStatus(getUrl, getModelId, getSubscriptionKey)
+
+      val convertTimeFormatUdf = UDFUtils.oldUdf(
+        { value: String => convertTimeFormat("Timestamp column", value) },
+        StringType
+      )
+      val formattedDF = dataset.withColumn(getTimestampCol, convertTimeFormatUdf(col(getTimestampCol)))
+        .sort(col(getTimestampCol).asc)
+        .withColumn("group", lit(1))
+
+      val window = Window.partitionBy("group").rowsBetween(-getBatchSize, 0)
+      var collectedDF = formattedDF
+      var columnNames = Array(getTimestampCol) ++ getInputVariablesCols
+      for (columnName <- columnNames) {
+        collectedDF = collectedDF.withColumn(s"${columnName}_list", collect_list(columnName).over(window))
+      }
+      collectedDF = collectedDF.drop("group")
+      columnNames = columnNames.map(name => s"${name}_list")
+
+      val testDF = getInternalTransformer(collectedDF.schema).transform(collectedDF)
+
+      testDF
+        .withColumn("isAnomaly", when(col(getOutputCol).isNotNull,
+          col(s"$getOutputCol.results.value.isAnomaly")(0)).otherwise(null))
+        .withColumn("DetectDataTimestamp", when(col(getOutputCol).isNotNull,
+          col(s"$getOutputCol.results.timestamp")(0)).otherwise(null))
+        .drop(columnNames: _*)
+
```

Comment:
The hard-coded part "{name}_list" is because I use collect_list to aggregate those values in different rows to a single row with a list of values. Only the suffix '_list' is hard-coded, and finally those columns will be dropped and values will be mapped back to the original dataframe.